client/signup/steps/plans/index.jsx

```diff
@@ -124,6 +124,13 @@ export class PlansStep extends Component {
 				themeSlugWithRepo: 'pub/lynx',
 			} );
 			this.props.goToNextStep();
+		} else if ( flowName === 'onboarding' && cartItem?.product_slug === 'ecommerce-bundle' ) {
+			// Buying ecommerce-bundle at signup gives you pub/twentytwentytwo
+			this.props.submitSignupStep( step, {
+				cartItem,
+				themeSlugWithRepo: 'pub/twentytwentytwo',
+			} );
+			this.props.goToNextStep();
```

Comment:
Hm, I can try this. I didn't do this at first because I thought the dependencies were specific to flows, and I didn't know how the other flows worked. But now I see they're specific to steps, so it should be safe to do inside the plans step regardless of context flow.